### PR TITLE
CSHARP-5226: Remove non-RenderArgs Render overloads

### PR DESCRIPTION
**Pull request by BorisDog**

### CODE REVIEW

File: src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToFilterTranslators/MethodTranslators/InjectMethodToFilterTranslator.cs [hunk 1]
```diff
@@ -39,13 +40,19 @@ public static AstFilter Translate(TranslationContext context, MethodCallExpressi
...
+                    renderArgsType,
+                    documentSerializer,
+                    serializerRegistry,
+                    default(PathRenderArgs),
+                    false, // renderDollarForm
+                    false, // renderForFind
+                    false); // renderForElemMatch
+
+                var renderMethod = filterDefinitionType.GetMethod(nameof(FilterDefinition<BsonDocument>.Render), new[] { renderArgsType });
+                var renderedFilter = (BsonDocument)renderMethod.Invoke(filterDefinition, new object[] { renderArgs });
```
Comment (BorisDog) [inline src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToFilterTranslators/MethodTranslators/InjectMethodToFilterTranslator.cs:55]: @rstam: Just discovered this dynamic rendering due to failing tests.

Comment (rstam) [reply src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToFilterTranslators/MethodTranslators/InjectMethodToFilterTranslator.cs:55]: I propose adding a new private static helper method:

```
// private static methods
private static BsonDocument RenderFilter<TDocument>(FilterDefinition<TDocument> filterDefinition, IBsonSerializer<TDocument> documentSerializer, IBsonSerializerRegistry serializerRegistry)
{
    return filterDefinition.Render(new(documentSerializer, serializerRegistry));
}
```

and replacing lines 44-55 with:

```
var renderFilterMethod = typeof(InjectMethodToFilterTranslator).GetMethod(nameof(RenderFilter), BindingFlags.Static | BindingFlags.NonPublic).MakeGenericMethod(documentType);
var renderedFilter = (BsonDocument)renderFilterMethod.Invoke(null, new object[] { filterDefinition, documentSerializer, serializerRegistry });
return AstFilter.Raw(renderedFilter);
```

This localizes the use of reflection to *this* file only and isolates us from (or at least turns into compile time errors) changes to the `RenderArgs` constructor or the signature of the `Render` method.

**Review by rstam — CHANGES_REQUESTED** [review]
Small change requested.

File: src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToFilterTranslators/MethodTranslators/InjectMethodToFilterTranslator.cs (hunk 1 above)
Comment (BorisDog) [reply src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToFilterTranslators/MethodTranslators/InjectMethodToFilterTranslator.cs]: Great idea! Done, thanks.

File: src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToFilterTranslators/MethodTranslators/InjectMethodToFilterTranslator.cs [hunk 2]
```diff
@@ -25,6 +26,13 @@ namespace MongoDB.Driver.Linq.Linq3Implementation.Translators.ExpressionToFilter
 {
     internal static class InjectMethodToFilterTranslator
     {
+        private readonly static MethodInfo __renderMethodInfo;
```
Comment (rstam) [inline src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToFilterTranslators/MethodTranslators/InjectMethodToFilterTranslator.cs:29]: I like this optimization of caching the MethodInfo in a static field.

But the field should be named `__renderFilterMethodInfo`.

File: src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToFilterTranslators/MethodTranslators/InjectMethodToFilterTranslator.cs [hunk 3]
```diff
@@ -37,19 +45,21 @@ public static AstFilter Translate(TranslationContext context, MethodCallExpressi
...
                 var documentSerializer = serializerRegistry.GetSerializer(documentType); // TODO: is this the right serializer?
-                var renderMethodArgumentTypes = new[]
-                {
-                    typeof(IBsonSerializer<>).MakeGenericType(documentType),
-                    typeof(IBsonSerializerRegistry)
-                };
-                var renderMethod = filterDefinitionType.GetMethod("Render", renderMethodArgumentTypes);
-                var renderedFilter = (BsonDocument)renderMethod.Invoke(filterDefinition, new object[] { documentSerializer, serializerRegistry });
+
+                var renderMethod = __renderMethodInfo.MakeGenericMethod(documentType);
```
Comment (rstam) [inline src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToFilterTranslators/MethodTranslators/InjectMethodToFilterTranslator.cs:52]: variable should be called `renderFiltermethod`

**Review by rstam — CHANGES_REQUESTED** [review]
Some naming fell through the cracks.

File: src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToFilterTranslators/MethodTranslators/InjectMethodToFilterTranslator.cs (hunk 2 above)
Comment (BorisDog) [reply src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToFilterTranslators/MethodTranslators/InjectMethodToFilterTranslator.cs]: Done.

File: src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToFilterTranslators/MethodTranslators/InjectMethodToFilterTranslator.cs (hunk 3 above)
Comment (BorisDog) [reply src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToFilterTranslators/MethodTranslators/InjectMethodToFilterTranslator.cs]: Done.

**Review by rstam — APPROVED** [review]
LGTM

**Comment by glen-84** [comment]
@BorisDog This is [another](https://github.com/mongodb/mongo-csharp-driver/pull/1393#issuecomment-2258348204) breaking change. Why was this released in a minor version?

**Pull request merged**